src/skillsaw/rules/builtin/hooks/dangerous.py

```
@@ -46,31 +49,29 @@
-        (?:&&|;)\s*                  # chained
-        {_SUDO}                      # optional sudo
-        (?:{_INTERPRETER_CMD})\s+\S+ # interpreter + file
+_DOWNLOAD_TOOL_RE = re.compile(r"\b(?:curl|wget)\b")
+# A fetch wrapped in process or command substitution feeds an interpreter
+# directly — `bash <(curl …)`, `bash -c "$(curl …)"` — with no shell
+# separator between the download and the interpreter, so the substitution
+# form is a download signal in its own right.
+_SUBSTITUTION_FETCH_RE = re.compile(
+    rf"""[<$]\(\s*(?:sudo\s+)?(?:curl|wget|nc|ncat)\b  # $(curl …)  <(curl …)
+        |`(?:sudo\s+)?(?:curl|wget|nc|ncat)\b          # `curl …`
     """,
```

Ruff (0.16.1)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)